apps/arweave/src/ar_block_pre_validator.erl

```diff
@@ -236,16 +224,6 @@ pre_validate_proof_sizes(B, PrevB, Peer) ->
 			invalid
 	end.
 
-validate_proof_size(PoA, Height) ->
```

Comment:
Not directly relevant, but a question I've had: is there a rule of thumb for knowing which code doesn't need to be aware of old fork heights?

Like any code that processes *new* blocks or transactions can remove the old fork checking code. But any code that needs to process *old* blocks or transactions still need to be aware of historical forks, right? (e.g. validating the full chain from genesis)

Are there certain modules (like ar_block_pre_vaidator?) that are only invoked for new blocks/transaction and other modules that might be invoked on really old blocks?

Reply:
I do not think we can draw a separation boundary on the module level. Yes, the code processing new blocks does not need to be aware of the old forks once the latest fork height drops below the checkpoint block (in practice, we usually keep it for some time before removing). `ar_block_pre_validator` is only concerned with the recent blocks, yes. The code that computes block hashes (`ar_block:indep_hash`) must be fork-aware as this is how the node verifies the integrity of the old blocks it downloads and asserts they belong to the current chain.

Reply:
Got it. Okay, so no rule of thumb I can follow. I'll just be careful with removing fork height code!